### PR TITLE
Use realm from current membership for emails

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* Use the current membership's realm for email templates
* Require UserWrite rbac for reset password

**Release Note**

```release-note
require UserWrite for admin reset of another user's password
```

### CODE REVIEW

**Review by mikehelmick — APPROVED** [review]

**Comment by mikehelmick** [comment]
compilation error

File: pkg/controller/login/reset_password.go [hunk 1]
```diff
@@ -59,6 +60,11 @@ func (c *Controller) HandleSubmitResetPassword() http.Handler {
 			return
 		}
 
+		membership := controller.MembershipFromContext(ctx)
+		if !membership.Can(rbac.UserWrite) {
+			controller.Unauthorized(w, r, c.h)
+			return
+		}
 		// Does the user exist?
```
Comment (sethvargo) [inline pkg/controller/login/reset_password.go:68]: Nit newline

File: pkg/controller/login/reset_password.go [hunk 2]
```diff
@@ -59,6 +60,11 @@ func (c *Controller) HandleSubmitResetPassword() http.Handler {
 			return
 		}
 
+		membership := controller.MembershipFromContext(ctx)
+		if !membership.Can(rbac.UserWrite) {
```
Comment (sethvargo) [inline pkg/controller/login/reset_password.go:64]: Doesn't this method also get called if the user resets their own password? That's why I didn't enforce rbac here.

File: pkg/controller/login/reset_password.go [hunk 3]
```diff
@@ -59,6 +60,11 @@ func (c *Controller) HandleSubmitResetPassword() http.Handler {
 			return
 		}
 
+		membership := controller.MembershipFromContext(ctx)
```
Comment (sethvargo) [inline pkg/controller/login/reset_password.go:63]: I don't think we require that a current realm is select for the user to reset their own password.

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**